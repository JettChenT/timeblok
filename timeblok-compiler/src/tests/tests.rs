use crate::{compile_deterministic, ir::ExactDateTime};
use insta::assert_snapshot;

macro_rules! assert_compile_snapshot {
    ($input:expr, $base_time:expr) => {
        assert_snapshot!(compile_deterministic($input, $base_time.clone()).unwrap());
    };

    ($input:expr) => {
        assert_compile_snapshot!($input, ExactDateTime::from_ymd_hms(2023, 1, 1, 0, 0, 0));
    };
}

#[test]
fn test_sanity() {
    assert_compile_snapshot!(r###"2023-4-4
10am wake up and eat breakfast
11am go to work
    "###);
}

#[test]
fn test_filters(){
    assert_compile_snapshot!(r###"2023--
{mon or tue or thu}
10am wakeup
{sat}
20:00 wekly review
    "###);

    assert_compile_snapshot!(r###"2024--
{workday}
6am wake up
{weekend}
10pm sleep
    "###);
}
