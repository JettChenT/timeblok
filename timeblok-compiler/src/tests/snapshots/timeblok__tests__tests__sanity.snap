---
source: timeblok-compiler/src/tests/tests.rs
expression: "compile_deterministic(r###\"2023-4-4\n10am wake up and eat breakfast\n11am go to work\n    \"###,\n        ExactDateTime::from_ymd_hms(2023, 1, 1, 0, 0, 0).clone()).unwrap()"
---
BEGIN:VCALENDAR
VERSION:2.0
PRODID:ICALENDAR-RS
CALSCALE:GREGORIAN
BEGIN:VEVENT
DTSTAMP:20230503T152958Z
DTEND:20230404T023000Z
DTSTART:20230404T020000Z
SUMMARY:wake up and eat breakfast
UID:c8e1eb22-a627-3daa-884c-2205cf78d075
END:VEVENT
BEGIN:VEVENT
DTSTAMP:20230503T152958Z
DTEND:20230404T033000Z
DTSTART:20230404T030000Z
SUMMARY:go to work
UID:9311744c-3746-3502-84c9-d06e8b5ea2d6
END:VEVENT
END:VCALENDAR

